cmd/site-replication.go

```diff
@@ -3058,3 +3272,995 @@ func (c *SiteReplicationSys) PeerEditReq(ctx context.Context, arg madmin.PeerInf
+		optsMap["location"] = bStatus.Location
+		opts.Location = bStatus.Location
+	}
+
+	optsMap["versioningEnabled"] = "true"
+	opts.VersioningEnabled = true
+	if bStatus.ObjectLockConfig != nil {
+		config, err := base64.StdEncoding.DecodeString(*bStatus.ObjectLockConfig)
+		if err != nil {
+			return err
+		}
+		if bytes.Equal([]byte(string(config)), enabledBucketObjectLockConfig) {
+			optsMap["lockEnabled"] = "true"
+			opts.LockEnabled = true
+		}
+	}
+	for _, dID := range dIDs {
+		peerName := info.Sites[dID].Name
+		if dID == globalDeploymentID {
+			err := c.PeerBucketMakeWithVersioningHandler(ctx, bucket, opts)
+			logger.LogIf(ctx, c.annotateErr("MakeWithVersioning", fmt.Errorf("error healing bucket for site replication %w from %s -> %s", err, latestPeerName, peerName)))
+			if err != nil {
+				return err
+			}
+		} else {
+			admClient, err := c.getAdminClient(ctx, dID)
+			if err != nil {
+				logger.LogIf(ctx, c.annotateErr("ConfigureRepl", fmt.Errorf("admin client not found: %w", err)))
+				return err
+			}
+			err = admClient.SRPeerBucketOps(ctx, bucket, madmin.MakeWithVersioningBktOp, optsMap)
+			logger.LogIf(ctx, c.annotatePeerErr(peerName, "MakeWithVersioning", err))
+			if err != nil {
+				return err
+			}
+			err = admClient.SRPeerBucketOps(ctx, bucket, madmin.ConfigureReplBktOp, nil)
+			logger.LogIf(ctx, c.annotatePeerErr(peerName, "ConfigureRepl", err))
+			if err != nil {
+				return err
+			}
+		}
+	}
+	if len(dIDs) > 0 {
+		// configure replication from current cluster to other clusters
+		err := c.PeerBucketConfigureReplHandler(ctx, bucket)
+		if err != nil {
+			logger.LogIf(ctx, c.annotateErr("ConfigureRepl", err))
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (c *SiteReplicationSys) healBucketReplicationConfig(ctx context.Context, objAPI ObjectLayer, bucket string, info srStatusInfo) error {
+	c.RLock()
+	defer c.RUnlock()
+	if !c.enabled {
+		return nil
+	}
+	bs, ok := info.BucketStats[bucket]
+	if !ok {
+		return nil
+	}
+	var replMismatch bool
+	for _, ss := range bs {
+		if ss.ReplicationCfgMismatch {
+			replMismatch = true
+			break
+		}
+	}
+	if replMismatch {
+		err := c.PeerBucketConfigureReplHandler(ctx, bucket)
+		if err != nil {
+			logger.LogIf(ctx, c.annotateErr("ConfigureRepl", err))
+			return err
+		}
+	}
+	return nil
+}
+
+func isBucketMetadataEqual(one, two *string) bool {
+	switch {
+	case one == nil && two == nil:
+		return true
+	case one == nil || two == nil:
+		return false
+	default:
+		return strings.EqualFold(*one, *two)
+	}
+}
+
+func (c *SiteReplicationSys) healIAMSystem(ctx context.Context, objAPI ObjectLayer) error {
+	info, err := c.siteReplicationStatus(ctx, objAPI, madmin.SRStatusOptions{
+		Users:    true,
+		Policies: true,
+		Groups:   true,
+	})
+	if err != nil {
+		return err
+	}
+	for policy := range info.PolicyStats {
+		c.healPolicies(ctx, objAPI, policy, info)
+	}
+
+	for user := range info.UserStats {
+		c.healLocalUsers(ctx, objAPI, user, info)
```

Comment:
It looks like we are not healing the service accounts here

Reply:
yes, we are not looks like its not supported.